### PR TITLE
Added documentation & Test Cases

### PR DESCRIPTION
**Pull request by karenrni**
Also removed unnecessary in & out models, rewrote & finished UI, and renamed entity and use case classes 

Still needs to connect with the chatroom

### CODE REVIEW

**Review by nina-huangg — APPROVED** [review]
nice!

File: src/test/java/studySpots/StudySpotUseCaseTest.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+import spots.useCases.RecsOutBoundary;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class StudySpotUseCaseTest {
+    @Test
+    public void testGenerateRecAllMutual() {
```
Comment (lancenigel) [inline src/test/java/studySpots/StudySpotUseCaseTest.java:16]: need to add java doc for public methods

File: src/main/java/spots/useCases/GenerateRec.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+        ArrayList<String> prefSpots1 = (ArrayList<String>) user1.getProfile().getStudySpotPreferences();
+        ArrayList<String> prefSpots2 = (ArrayList<String>) user2.getProfile().getStudySpotPreferences();
+
+        spotRecommender.setPrefSpots(prefSpots1, prefSpots2);
+        spotRecommender.recommender();
+        return spotRecommender.getRecommendation();
+    }
+
+    /**
```
Comment (lancenigel) [inline src/main/java/spots/useCases/GenerateRec.java:45]: i think javadoc should be within the method fr example:
    public void createRecs() {
    /**
     * Create a recommendation
     * Update the user interface with the recommendation
     */
        //RecsOutModel recModel = new RecsOutModel(generateRec());
        recsPresenter.update(generateRec());
    }

**Review by tankenji — DISMISSED** [review]

**Review by SanzSS — APPROVED** [review]

**Review by tankenji — APPROVED** [review]

**Review by SanzSS — COMMENTED** [review]
Nice test cases! Hoping its high coverage!

In terms of CA, everything seems to be spotless, I dont spot any dependency issues for example (dependencies are mostly correct in fact, such as the controller depending on the input boundary). Interfaces are made and implemented too, such GenerateRec implementing RecsInBoundary.

In terms of SOLID. S is achieved as none of the classes seem to be doing more than their purpose. O is achieved with your good use of interfaces. L is achieved, but this is technically more so because you don't have many (any) subclasses. I believe I is achieved, but this is a bit harder to tell. I can't see any interfaces being implemented by many so I think you're good. D is achieved by making sure a number of your classes depend on abstractions (interfaces), such as how GenerateRec implements on RecsInBoundary.

No code smells! I especially like how well organized your test codes are!

**Pull request merged**